frontend-react/src/hooks/UseCreateFetch.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback } from "react";
+import { AccessToken } from "@okta/okta-auth-js";
+import axios from "axios";
+
+import { RSEndpoint, AxiosOptionsWithSegments } from "../config/endpoints";
+
+import { MembershipSettings } from "./UseOktaMemberships";
+
+export type AuthorizedFetcher<T> = (
+    EndpointConfig: RSEndpoint,
+    options?: Partial<AxiosOptionsWithSegments>
+) => Promise<T>;
+
+// this wrapper is needed to allow typing of the fetch return value from the location
+// where the hook is being called. If we returned an AuthorizedFetcher directly from the hook
+// there wouldn't be a way to type it from the location calling the hook, as that would
+// require typing the `useContext` call in a way that useContext doesn't support
+export type AuthorizedFetchTypeWrapper = <T>() => AuthorizedFetcher<T>;
+
+// takes in auth data and returns
+//  a generic function that returns
+//    a function that can be used to make an API call
+function createTypeWrapperForAuthorizedFetch(
+    oktaToken: Partial<AccessToken>,
+    activeMembership: MembershipSettings
+) {
+    const authHeaders = {
```

Comment:
add a comment to the effect that all requests will expect auth headers to be present. maybe log on the request if we're missing headers